packages/host/app/components/operator-mode/card-url-bar.gts

```diff
@@ -0,0 +1,190 @@
+        <div class='error-message' data-test-card-url-bar-error>
+          <span>{{if this.isInvalid this.errorMessage @errorMessage}}</span>
+        </div>
+      {{/if}}
+    </div>
+    <style>
+      .card-url-bar {
+        display: flex;
+        align-items: center;
+
+        background: var(--boxel-purple-700);
+        border-radius: var(--boxel-border-radius-xl);
+        padding: var(--boxel-sp-xs) 0 var(--boxel-sp-xs) var(--boxel-sp-sm);
+
+        width: 100%;
+        position: relative;
+      }
+      .focused {
+        outline: 2px solid var(--boxel-teal);
+      }
+      .invalid {
+        outline: 2px solid red;
+      }
+      .realm-info {
+        display: flex;
+        align-items: center;
+        gap: var(--boxel-sp-xs);
+
+        width: max-content;
+        color: var(--boxel-light);
+        border-right: 2px solid var(--boxel-purple-300);
+        padding-right: var(--boxel-sp-xs);
+        margin-right: var(--boxel-sp-xs);
+
+        white-space: nowrap;
+      }
+      .realm-info img {
+        width: 22px;
+      }
+      .input {
+        display: flex;
+        align-items: center;
+        gap: var(--boxel-sp-xs);
+        width: 100%;
+
+        --icon-color: var(--boxel-cyan);
+      }
+      .url-input {
+        background: none;
+        border: none;
+        border-radius: 0;
+        outline: none;
+        padding: 0;
+        min-height: 0;
+
+        color: var(--boxel-light);
+      }
+      .error-message {
+        position: absolute;
+        bottom: calc(calc(var(--boxel-sp-xs) * 2) * -1);
+        color: red;
+      }
+    </style>
+  </template>
+
+  @service declare cardService: CardService;
+  @tracked isFocused: boolean = false;
+  @tracked url: string = this.args.url.toString();
+  @tracked isInvalid: boolean = false;
+  @tracked errorMessage: string | null = null;
+
+  get realmIcon() {
+    return this.fetchRealmInfo.value?.iconURL;
+  }
+
+  get realmName() {
+    return this.fetchRealmInfo.value?.name;
+  }
+
+  get cardDisplayName() {
+    if (!this.args.card) return;
+    return cardTypeDisplayName(this.args.card);
+  }
+
+  get showErrorMessage() {
+    return (
+      (this.args.isInvalid && this.args.errorMessage) ||
+      (this.isInvalid && this.errorMessage)
+    );
+  }
+
+  get cssStyle() {
+    if (this.args.isInvalid) {
+      return 'card-url-bar invalid';
+    } else if (this.isFocused) {
+      return 'card-url-bar focused';
+    } else {
+      return 'card-url-bar';
+    }
+  }
+
+  fetchCard = trackedFunction(
+    this,
+    async () => await this.cardService.loadModel(this.args.url),
+  );
+
+  fetchRealmInfo = trackedFunction(this, async () => {
+    if (!this.args.card) return;
+    return this.cardService.getRealmInfo(this.args.card);
+  });
+
+  @action
+  onInput(url: string) {
+    this.url = url;
+  }
+
+  @action
+  onKeyPress(event: KeyboardEvent) {
+    try {
+      if (event.key === 'Enter' || event.keyCode === 13) {
```

Comment:
Is it necessary to use both conditions? Does only the `event.key === 'Enter'` not suffice?

Reply:
confirm @jurgenwerk's suggestion is the modern way to do this